### DISCUSSION TITLE
Combine pictures and save as searchable pdf with desired language with own directory names

### DISCUSSION THREAD

**Discussion by kazimates** [py-pdf/pypdf#1734] [Q&A] [answered]
How can I combine multiple images under different folders with a folder name? But the pdf files should be searchable pdf in desired language ie eng, fra, tur which is set in python code. I try to write the code with instructions on the internet or ChatGPT but non of them worked. Can you help to solve this problem, please?

the pictures are located C:\Users\kazim.ates\Desktop\Newsletter\FullPages\20230319\ and I have 100 directories and all of the directories have pictures in them.
The pdf files should be saved into C:\Users\kazim.ates\Desktop\Newsletter\FullPagesPDF\20230319\

**Comment by pubpub-zz** [comment]
I'm really confused about your question :
You want to create a single pdf with many pages where each page is a image(JPEG?) in a folder. => OK
you are then talking about searchable PDF. Where is the text coming from? where will it be visble?

**Comment by kazimates** [comment]
the text is on the picture. Otherwise scan one newspaper. It is 40 pages and every page is scanned in a jpg format. the pages scanned ie 001.jpg, 002.jpg .... 040.jpg and it is into one directory ie c:\newspapers\20220323\20230323_newspapername\ 
now how can I combine all jpg images under c:\newspapers\20220323\20230323_newspapername\ with a name 20230323_newspapername.pdf but it this pdf should be searchable pdf as a language ie fra, tur or which one need it. The searchable pdf will be save under c:\newspapers\20220323\20230323_newspapername.pdf
I can't solve this problem.

**Reply by MartinThoma** [reply]
Please ask only one question at a time. We are not there to write a complete project for you: https://stackoverflow.com/help/how-to-ask gives a good insight how to ask good questions.

**Comment by pubpub-zz** [comment, marked as answer]
pypdf does not include any OCR => extract text from images will not be possible
to convert jpg to pdf, I recommend you to use PIL and then use pypdf to assemble them

as said we are some volunteers. at least you should have look at  #1660
